src/lib/lib.c

```diff
@@ -67,26 +70,36 @@ void *memcpy(void *dest, const void *src, size_t n)
 	return dest;
 }
 
-/* generic bzero - TODO: can be optimsed for ARCH ? */
-void bzero(void *s, size_t n)
+int memcpy_s(void *dest, size_t dest_size, const void *src, size_t src_size)
 {
-	uint32_t *d32 = s;
-	uint8_t *d8;
-	int i;
-	int d = n >> 2;
-	int r = n % 4;
+	return arch_memcpy_s(dest, dest_size, src, src_size);
+}
 
-	/* zero word at a time */
-	for (i = 0; i <	d; i++)
-		d32[i] = 0;
+int memset_s(void *dest, size_t dest_size, int data, size_t count)
+{
+	if (!dest)
+		return -EINVAL;
 
-	/* zero remaining bytes */
-	d8 = (uint8_t*) &d32[i];
-	for (i = 0; i <	r; i++)
-		d8[i] = 0;
+	if (count > dest_size)
+		return -EINVAL;
+
+#if __XCC__ && !CONFIG_HOST
```

Comment:
Should not be adding xtensa _XCC_ to generic code. This should be abstracted like the others.